### PR TITLE
Feature/sonarqube code smells

### PR DESCRIPTION
**Pull request by theo-ritense**

### CODE REVIEW

File: contract/src/main/java/com/ritense/valtimo/contract/security/jwt/TokenAuthenticator.java [hunk 1]
```diff
@@ -19,10 +19,10 @@
 import io.jsonwebtoken.Claims;
 import org.springframework.security.core.Authentication;
 
-public abstract class TokenAuthenticator {
+public interface TokenAuthenticator {
```
Comment (marijnritense) [inline contract/src/main/java/com/ritense/valtimo/contract/security/jwt/TokenAuthenticator.java:22]: Don't forget to also document this as a breaking change

File: core/src/main/java/com/ritense/valtimo/camunda/repository/CustomRepositoryServiceImpl.java [hunk 2]
```diff
@@ -35,11 +35,6 @@ public CustomRepositoryServiceImpl(final ApplicationEventPublisher applicationEv
         this.applicationEventPublisher = applicationEventPublisher;
     }
 
-    @Override
-    public DeploymentWithDefinitions deployWithResult(DeploymentBuilderImpl deploymentBuilder) {
```
Comment (marijnritense) [inline core/src/main/java/com/ritense/valtimo/camunda/repository/CustomRepositoryServiceImpl.java:39]: Why was this removed?

Comment (theo-ritense) [reply core/src/main/java/com/ritense/valtimo/camunda/repository/CustomRepositoryServiceImpl.java:39]: Because it overrides a method and then only calls the `super`.

Comment (theo-ritense) [reply core/src/main/java/com/ritense/valtimo/camunda/repository/CustomRepositoryServiceImpl.java:39]: Because it overrides a method, but only calls `super`. So nothing changed effectively.

File: contract/src/main/java/com/ritense/valtimo/contract/security/jwt/TokenAuthenticator.java (hunk 1 above)
Comment (theo-ritense) [reply contract/src/main/java/com/ritense/valtimo/contract/security/jwt/TokenAuthenticator.java:22]: I made a not of it in the story.

**Review by marijnritense — APPROVED** [review]
